docs/web/howto/client.rst

```diff
@@ -331,74 +331,85 @@ This function returns a ``Deferred`` that fires with the body after the request
+        def gotResponse(response):
+            print(response.code)
+        def noResponse(failure):
+            failure.trap(ResponseFailed)
+            print(failure.value.reasons[0].getTraceback())
+        return requested.addCallbacks(gotResponse, noResponse)
 
+This is because the certificate returned by ``wrong.host.badssl.com`` is actually for ``badssl.com``, and the hostnames don't match.
+In our pretend scenario, we want to construct an ``Agent`` that validates HTTPS certificates normally, _except_ for this one host, where it ignores the hostname mismatch and supplies the correct hostname manually, to work around this problem.
+In order to do that, we will supply our own policy that creates a per - netloc–i.e. per host+port combination – TLS configuration, like so:
 
-For more fine-grained over the TLS configuration, check out the documentation for the :api:`twisted.internet.ssl <ssl>` module and :doc:`Using SSL in Twisted <../../core/howto/ssl>`.
+.. code-block:: python
 
+    import sys
 
+    from zope.interface import implementer
 
+    from twisted.internet.task import react
+    from twisted.internet.ssl import optionsForClientTLS
 
+    from twisted.web.iweb import IPolicyForHTTPS
+    from twisted.web.client import Agent, ResponseFailed, BrowserLikePolicyForHTTPS
+
+    @implementer(IPolicyForHTTPS)
+    class OneHostnameWorkaroundPolicy(object):
+        def __init__(self):
+            self._normalPolicy = BrowserLikePolicyForHTTPS()
+        def creatorForNetloc(self, hostname, port):
+            if hostname == b"wrong.host.badssl.com":
+                return optionsForClientTLS(u"badssl.com")
+            else:
+                return self._normalPolicy.creatorForNetloc(hostname, port)
+
+    @react
+    def main(reactor):
+        agent = Agent(reactor, OneHostnameWorkaroundPolicy())
+        requested = agent.request(b"GET", sys.argv[1])
+        def gotResponse(response):
+            print(response.code)
+        def noResponse(failure):
+            failure.trap(ResponseFailed)
+            print(failure.value.reasons[0].getTraceback())
+        return requested.addCallbacks(gotResponse, noResponse)
+
+Now, invoking ``python example.py https://wrong.host.badssl.com/`` will happily give us a ``200`` status code; however, running it with ``https://expired.badssl.com/`` or ``https://self-signed.badssl.com/`` or any of the other error hostnames should still give an error.
+
+Using this TLS policy mechanism, you can customize Agent to use any feature of TLS that Twisted has support for, including the examples given above; client certificates, alternate trust roots, and so on.
+For a more detailed explanation of what options exist for client TLS configuration in Twisted, check out the documentation for the :api:`twisted.internet.ssl.optionsForClientTLS <optionsForClientTLS>` API and the :doc:`Using SSL in Twisted <../../core/howto/ssl>` chapter of this documentation.
```

Comment:
It might be nice to standardize on a "Further Reading" section in the docs which contains tips like this one (providing users with the benefit of navigational simplicity in accessing the information).